File: easiness-ui/app/server/entity/product.ts
import { Entity, PrimaryGeneratedColumn, Column } from "typeorm"

@Entity()
export class Product {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    type: string

    @Column()
    brand: string

    @Column()
    size: string
    
    @Column()
    country: string

}

File: easiness-wails/internal/models/product.go
package models

type Product struct {
	Base
	Name          string `gorm:"not null;uniqueIndex:idx_product_unique" json:"name"`
	Type          string `gorm:"uniqueIndex:idx_product_unique"          json:"type"`
	Brand         string `gorm:"uniqueIndex:idx_product_unique"          json:"brand"`
	Country       string `gorm:"not null"                                json:"country"`
	Size          string `gorm:"uniqueIndex:idx_product_unique"          json:"size"`
	PreferredUnit uint   `gorm:"not null"                                json:"preferredUnit"`
}
